orbit/eda/eda_plot.py

```diff
@@ -0,0 +1,206 @@
+    sns.lineplot(data=df, x=date_col, y=outcome, ax=ax[1])
+    ax[1].legend(handle_list, label_list, loc='center left', bbox_to_anchor=(1, 0.5), fancybox=True, shadow=True)
+    ax[1].title.set_text(f'{outcome} Time Series with Events Plot')
+
+    if export_fig:
+        plt.savefig(f'{outcome} and Events Year Over Year Plot.pdf')
+        print('plot_exported')
+
+
+def dual_axis_ts_plot(df, var1, var2, date_col, fig_width=25, fig_height=6, export_fig=False):
+    """
+    This function plots two time series variables on two y axis. This is handy for comparison of two variables. The dual
+    y axis will set on two different scales if the two variables are very different in terms of volume.
+    :param df: input time series dataframe
+    :param var1: the first var to be plotted
+    :param var2: the second var to be plotted
+    :param date_col: date column
+    :param fig_width: optional, adjust width of the chart
+    :param fig_height: optional, adjust height of the chart
+    :param export_fig: optional, export chart to the same folder
+    :return: one time series plot with dual y axis
+    """
+    df[date_col] = pd.to_datetime(df[date_col])
+    fig, ax = plt.subplots(figsize=(fig_width, fig_height))
+    sns.lineplot(data=df, x=date_col, y=var1, label=var1, color='#4c72b0')
+    ax.set_ylabel(f'{var1}', color='#4c72b0')
+    ax.tick_params(axis='y', labelcolor='#4c72b0')
+    ax2 = ax.twinx()
+    sns.lineplot(data=df, x=date_col, y=var2, ax=ax2, color='#dd8452', label=var2)
+    ax2.set_ylabel(f'{var2}', color='#dd8452')
+    ax2.tick_params(axis='y', labelcolor='#dd8452')
+    lines, labels = ax.get_legend_handles_labels()
+    lines2, labels2 = ax2.get_legend_handles_labels()
+    ax2.legend(lines + lines2, labels + labels2, loc=0)
+    ax.set_title(f'{var1} vs {var2}')
+    # ax.grid(False)
+    ax2.grid(False)
+
+    if export_fig:
+        plt.savefig(f'{var1} vs {var2}.pdf')
+        print('plot_exported')
+
+
+def wrap_plot_ts(df, date_col, var_list, col_wrap=3, height=2.5, aspect=2):
+    non_zero_varlist = [i for i in var_list if i not in df.columns[(df == 0).all()]]
+    df = df[non_zero_varlist]
+    df_long = df.melt(id_vars=[date_col])
+    g = sns.relplot(x=date_col, y='value', col='variable', height=height, aspect=aspect,
+                col_wrap=col_wrap, kind='line', data=df_long, facet_kws={'sharey': False, 'sharex': False})
+    g.set_xticklabels(rotation=45)
+    plt.tight_layout()
+
+
+def weekly_trend_decomposition(df, var, date_col):
+    '''
+    This function presents variable weekly trend after removing weekly seasonality.
+    :param df: input time series dataframe
+    :param var: variable need to be plotted
+    :param date_col: specify the timestamp for var
+    :return: three time series plots: Trend, Residual and Weekly Seasonality
+    '''
+    df_dt = deepcopy(df)
+    df_dt.index = pd.to_datetime(df_dt[date_col])
+    res_weely = seasonal_decompose(df_dt[[var]], period=7, model='multiplicative')
```

Comment:
I am trying to generate some EDA plots so looking at this code to borrow some ideas. A quick question, where do we get this seasonal_decompose(). Thanks!